security/RFP.md

```diff
@@ -0,0 +1,254 @@
+
+Responder can formulate comments about the above hypothesis. Especially if they can suggest less restricting hypothesis under which the security properties of Hydra still hold. Or if they would suggest practical suggestion on how to implement this hypothesis. We would accept recommandations about how to ensure this hypotheses on a running environment.
+
+## Tasks
+
+On a broad level, our goal is to ensure that the security properties proven in the original publication hold for the implementation, taking also into consideration the main entry points of a hydra node which are the network, the API and the Cardano ledger.
+
+TODO: say something about increasing confidence to our users in using the hydra-node and the protocol implemented by it;
+
+
+Given the artifacts described above, 
+We expect the auditor to assess several statements which will be detailed in the following sections.
+
+By order of priority:
+-- pen and paper stuff
+1. Coordinated Hydra Head V1 specification is consistent with the original publication
+
+TODO: ask Yun what she thinks about that
+
+--plutus stuff
+
+2. Hydra plutus scripts are consistent with Hydra Head v1 specification
+3. Hydra plutus scripts are immune to common Cardano smart-contract weaknesses
+
+-- hydra node implementation is behaving as an honest actor (out of memory denial of service...)
+
+4. Hydra node chain layer code generates transactions which are consistent with Hydra Head v1 specification
+5. Hydra node logic layer code is consistent with Hydra Head v1 specification
+
+-- robust against the environment whatever
+
+6. Hydra head protocol implementation is immune to attacks via chain transactions
+7. Hydra head protocol implementation is immune to attacks via network
+8. Hydra head protocol implementation faithfully reflects the head state through the API
+
+Out os scope:
+1. Hydra head protocol implementation is immune to API attacks -- out of scope because trusted
+
+FIXME: split the RFP or give the option to respond partially. Ask for a detailed estimation of each of the tasks so that we may choose to not ask for everything.
+FIXME: we want the respondant to explain what kind of analysis they will use for each of the assessments.
+
+### Coordinated Hydra Head V1 specification is consistent with the original publication
+
+The Hydra Head v1 specification describes Coordinated Hydra Head V1 Protocol.
+
+This specification provides several important security proofs, especially that the following four properties hold:
+
+* Consistency: No two uncorrupted parties see conflicting transactions confirmed.
+* Liveness: If all parties remain uncorrupted and the adversary delivers all messages, then every transaction becomes confirmed at some point.
+* Soundness: The final UTxO set accepted on the mainchain results from a set of seen transactions.
+* Completeness: All transactions observed as confirmed by an honest party at the end of the protocol are considered on the mainchain.
```

Comment:
This is repeated from above. Overall I think the RFP/scope contains a lot of repetition. It's tiring to read.

Reply:
I resolve this one because I feel like you already fixed that. If I'm wrong, please re-open.